lambdas/ec2/reaper.py

```diff
@@ -88,9 +91,19 @@ def wait_for_tags(ec2_instance, wait_time):
         if not lifetime_match:
             terminate_instance(ec2_instance, 'Invalid lifetime value supplied')
             return
-        lifetime_delta = calculate_lifetime_delta(lifetime_match)
-        future_termination_date = start + lifetime_delta
-        ec2_instance.create_tags(
+        if lifetime_match == INDEFINITE:
```

Comment:
I think that the `if not lifetime_match` conditional above this will make this logic not work; that is, the `lifetime` value of "indefinite" will return `None` for `lifetime_match` and consequently terminate the instance. But, I believe you said you had this up and running in an account; have you seen this behavior work there?

Reply:
Oh, I see now, the `validate_lifetime_value` method is changed to handle the 'indefinite' value.

Reply:
Yeah I added 'indefinite' as a legitimate value in the validate_lifetime_value() function. If it gets a lifetime of 'indefinite' it returns that same value to tag the instance with. 

It is running in the us-west-1 region of an AWS account and this behavior is currently working there.